### DISCUSSION TITLE
npm package use only-allow

### DISCUSSION THREAD

**Discussion by xiaoxiaojx** [pnpm/pnpm#4131] [Q&A] [answered]
An npm package scripts with "preinstall": "npx only-allow pnpm",
will affect the projects that actually use that npm package

So I want to add a judgment to the only-allow
![image](https://user-images.githubusercontent.com/23253540/146132970-92ecb42d-fac2-4373-b922-403a39958b44.png)

**Comment by zkochan** [comment, marked as answer]
I am not sure this fix will work. But if it will, sounds good.

**Reply by xiaoxiaojx** [reply]
The above method is wrong, and finally I adopted the following method, so that the project that actually depends on the npm package is not affected 
https://github.com/xiaoxiaojx/only-allow/commit/88dec179c6302a844cf9782f11ae798be7597d13
I can mention pr if needed ~

**Reply by xiaoxiaojx** [reply]
Now our project temporarily uses the only-allow-test package, if anyone else encounters this problem can refer to it, first marked as solved
